ckanext/oidc_pkce/plugin.py

```diff
@@ -40,10 +53,46 @@ def get_helpers(
     ):
         return helpers.get_helpers()
 
-    if not tk.check_ckan_version("2.10"):
-        p.implements(p.IAuthenticator, inherit=True)
+    # IAuthenticator
+
+    if tk.check_ckan_version("2.10"):
+
+        def logout(self):
+            """ We want to return a view after the regular logout logic,
+            rather than before.
+
+            We set a flag to indicate that we're in the middle of logout,
+            then call the regular logout view. The view calls a second
+            instance of this function, which detects the flag and no-ops,
+            allowing the view to proceed and wipe the session.
+
+            After it completes, we assemble a redirect and pass that back
+            to the code that originally called this function.
+            """
+            if session.pop("_in_logout", False):
+                log.debug("SSO logout found in-progress flag, skipping recursive call")
+                return None
+            current_user = _current_user()
+            if not current_user.is_authenticated:
+                log.info("No current user found, skipping SSO logout")
+                return None
+            plugin_extras = getattr(current_user, 'plugin_extras', None)
+            if not plugin_extras or not plugin_extras.get('oidc_pkce'):
+                log.info("Current user [%s] is not associated with SSO, skipping SSO logout",
+                         current_user.name)
+                return None
 
-        # IAuthenticator
+            log.info("Logging out [%s]", current_user.name)
+            sso_logout_url = config.logout_url()
+            if not sso_logout_url:
+                log.info("No SSO logout path configured, logout of [%s] will be local only",
+                         current_user.name)
+                return None
+            session["_in_logout"] = True
```

Comment:
Can you confirm that our hosting will send them back to the same server ( or refresh me that we have a Memcache shared between the 5 server instances so it will work) am a bit fuzzy on this

Reply:
Our load balancer is configured with cookie-based stickiness: https://github.com/qld-gov-au/ckan-qld-infrastructure/blob/a441ea7711b326d7d4970a59a1626ee4b017a744/templates/Datashades-OpsWorks-CKAN-Stack.cfn.yml.j2#L596